src/utils/logging_utils.py

```
@@ -0,0 +1,460 @@
+        self.performance_history = deque(maxlen=1000)  # Keep last 1000 operations
+        
+        # Thread safety
+        self._metrics_lock = threading.Lock()
+    
+    def _setup_structured_logging(self):
+        """Setup structured logging format."""
+        # Create custom formatter for structured logs
+        class StructuredFormatter(logging.Formatter):
+            def format(self, record):
+                log_entry = {
+                    "timestamp": datetime.fromtimestamp(record.created, tz=timezone.utc).isoformat(),
```

Ruff (0.12.2)

Use `datetime.UTC` alias

Convert to `datetime.UTC` alias

(UP017)